### ISSUE TITLE
Beginner Question

### ISSUE THREAD

**Issue by Sp3cOz** [jmar777/kwicks#31]
Hello

I'm somewhat of a coding novice, I'm trying to implement kwicks into a site I'm working on, however, on first load, the kwicks menu stretches across the entire browser window, as soon as I mouse over the menu, it reverts to the correct width of my page container.

![menu](https://f.cloud.github.com/assets/5651649/1302982/07f65f0c-3144-11e3-9b91-44647068818a.jpg)

I suspect this has to do with the autoResize function but I have no idea how to fix it.

Thanks

**Comment by jmar777** [comment]
Hey - would you mind posting the Kwicks options you're using?

**Comment by Sp3cOz** [comment]
This is in the head section

```
<style type='text/css'>
            .kwicks {
                width: 1200px;
                height: 25px;
            }
            .kwicks > a {
                height: 25px;
                width: 297px;
                padding: 0;
                margin-left: 3px;
                float: left;
            }

            #panel-1 { background:url(images/menubghome.jpg) no-repeat; text-decoration:none; font-size:12px; text-align:left; line-height:25px; color:white; text-indent:9px;}
            #panel-2 { background:url(images/menubgbuild.jpg) no-repeat; text-decoration:none; font-size:12px; text-align:left; line-height:25px; color:white; text-indent:9px;}
            #panel-3 { background:url(images/menubgcomplete.jpg) no-repeat; text-decoration:none; font-size:12px; text-align:left; line-height:25px; color:white; text-indent:9px;}
            #panel-4 { background:url(images/menubgcontact.jpg) no-repeat; text-decoration:none; font-size:12px; text-align:left; line-height:25px; color:white; text-indent:9px;}
         </style>
```

And this is in my page container after the anchors

```
<script type='text/javascript'>
            $().ready(function() {
                $('.kwicks').kwicks({
                    maxSize : 450,
                    behavior: 'menu'
                });
            });
        </script>
```

I've basically just tried to set it up using the example page and customise from there, no extra options were added.

**Comment by jmar777** [comment]
Hey - sorry for the delay on this, I was out of town for a bit. Are you still having any issues here? Nothing obvious jumps out at me from your code, so I think I would need to see the actual page itself.

**Comment by mad93** [comment]
I had the same issue, just added a resize after initialization:

```
$(function() {
    $('.kwicks').kwicks({
        maxSize : 100,
        behavior: 'menu',
        isVertical: true
    });

    $('.kwicks').kwicks('resize');

});
```

**Comment by jmar777** [comment]
Interesting. If anyone can create a reproducible test case for this, I'll get a fix in for it. I just haven't been able to reproduce this on my own...

**Comment by CFusion-zz** [comment]
I am having a likewise problem, it fixes itself upon the re-sizing of the window.

Calling resize of the element does nothing.

It only seems to happen in chrome, and only ~10% of the time.

Edit:

![2013-10-29_21-59-56](https://f.cloud.github.com/assets/1655727/1432378/a9bd26d8-40dd-11e3-818a-a590bb07e5cb.png)

**Comment by jmar777** [comment]
Well, that's fun. I'll spend some more time trying to reproduce this. There might just need to be another call to `updatePanelStyles()` in certain anomalous situations...

**Comment by jmar777** [comment]
Alright, after several unsuccessful attempts at recreating the problem, I just pushed out a patch release with a blind stab at a fix (v2.2.1).  Given that the problem resolves itself when you mouseover, my guess is there's a rare timing issue that sometimes occurs on page load.  The update simply adds another call to `updatePanelStyles()` inside a `setTimeout()`.

I'm going to (optimistically) close this issue for now, but please feel free to reopen if you continue to see the problem in the new version.  Also, if it does seem to fix the problem, I'd love to hear back with confirmation :)

**Comment by ghost** [comment]
I have an identical problem, in v2.2.1, on Firefox.

I found that the first calculation of the "panel size", around line 308, was wrong because it was initially based on a much too large "container size" (perhaps one of the parent containers?), which is itself derived from the "primary dimension".

Opening and closing Firebug (or resizing the window) is enough to cause the correct calculation of the panel size.

[Edit: I thought using kwicks('resize') also resolved it, but apparently not.]
